The sirius cross referencer does not support derived reference

version sirius-component 2022.9.1

* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- latest
- other: please specify
* [ ] **I have specified my environment** : 
- All
- my computer: Windows? Linux? Cloud?
- Cloud

### Screenshots

...

### Steps to reproduce

* cerate a document in a project. The document must contain objects referenced from a derived reference
* delete the document

### Expected behavior

the document should delete

### Actual behavior

depending on the implementation of the modeler we may get an exception 
Example for uml2
```
Caused by: java.lang.UnsupportedOperationException: null
	at org.eclipse.uml2.common.util.DerivedEObjectEList$DerivedListIterator.remove(DerivedEObjectEList.java:273) ~[org.eclipse.uml2.common-2.5.0-SNAPSHOT.jar:na]
	at java.base/java.util.AbstractCollection.remove(AbstractCollection.java:299) ~[na:na]
	at org.eclipse.emf.ecore.util.EcoreUtil.remove(EcoreUtil.java:3228) ~[org.eclipse.emf.ecore-2.23.0.jar:na]
	at org.eclipse.sirius.components.emf.services.EditingContextCrossReferenceAdapter.clearReferencesTo(EditingContextCrossReferenceAdapter.java:96) ~[classes/:na]
	at org.eclipse.sirius.components.emf.services.EditingContextCrossReferenceAdapter.handleRemoveEObject(EditingContextCrossReferenceAdapter.java:63) ~[classes/:na]
```

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -8,19 +8,22 @@
 - [ADR-070] How to contribute a new diagram node style
 - [ADR-071] Add parametric SVG node style
 - [ADR-072] Add support for the edition of routing points
-- [ADR-072] Add support for intermodel references (inside the same project)
+- [ADR-073] Add support for intermodel references (inside the same project)
  
 === Breaking changes
 
 - https://github.com/eclipse-sirius/sirius-components/issues/1300[#1300] [core] Rename SiriusWebJSONResourceFactoryImpl to JSONResourceFactoryImpl
+- https://github.com/eclipse-sirius/sirius-components/issues/1301[#1301] [core] Add support for having intermodel references. Now, the URI of resources added in a ResourceSet must have a `scheme`. Otherwise, you will  encounter exceptions.
 
+=== Dependency update
+
+- [backend] Switch to EMFJson 2.3.2-SNAPSHOT. It contains better management of proxy resolution in case of intermodel references.
 
 === Bug fixes
 
-- https://github.com/eclipse-sirius/sirius-components/issues/1301[#1301] [core] Add support for having intermodel references
 - https://github.com/eclipse-sirius/sirius-components/issues/1368[#1368] [diagram] Do not render rectangle view children twice
 - https://github.com/eclipse-sirius/sirius-components/issues/931[#931] [layout] When the delete tool of an edge is used, the position of other edges are not affected.
-
+- https://github.com/eclipse-sirius/sirius-components/issues/1392[#1392] [core] Fix EditingContextCrossReferenceAdapter for derived references
 
 === Improvements
 
@@ -75,7 +78,6 @@ The `label` value is displayed before the widget to be consistent with other pro
 - [frontend] Switch to `@apollo/client 3.6.9`, `@material-ui/core 4.12.4`, `@material-ui/icons 4.11.3`, `graphql 16.5.0`, `prop-types 15.8.1`, `sprotty 0.12.2` and `xstate 4.32.1`. Various development dependencies have also been updated such as typescript, rollup, prettier etc
 - [backend] Switch to https://github.com/spring-projects/spring-boot/releases/tag/v2.7.2[Spring Boot 2.7.2]
 - [backend] Switch to the managed version of GraphQL Java, as of Spring Boot 2.7.2, we will thus use GraphQL Java 18.2
-- [backend] Switch to EMFJson 2.3.2-SNAPSHOT
 
 === Bug fixes
 

--- a/doc/adrs/073_add_support_for_intermodel_reference.adoc
+++ b/doc/adrs/073_add_support_for_intermodel_reference.adoc
@@ -1,4 +1,4 @@
-= ADR-072 - Add support for intermodel references (inside a same project)
+= ADR-073 - Add support for intermodel references (inside a same project)
 
 == Context
 

--- a/packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/EditingContextCrossReferenceAdapter.java
+++ b/packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/EditingContextCrossReferenceAdapter.java
@@ -103,16 +103,16 @@ public class EditingContextCrossReferenceAdapter extends ECrossReferenceAdapter 
     }
 
     private Collection<Setting> getNonContainmentReferences(Collection<Setting> inverseReferences) {
-        Collection<Setting> containmentReferences = new ArrayList<>();
+        Collection<Setting> nonContainmentReferences = new ArrayList<>();
         for (EStructuralFeature.Setting setting : inverseReferences) {
             EStructuralFeature eStructuralFeature = setting.getEStructuralFeature();
             if (eStructuralFeature instanceof EReference) {
                 EReference eReference = (EReference) eStructuralFeature;
-                if (!eReference.isContainment()) {
-                    containmentReferences.add(setting);
+                if (!eReference.isContainment() && !eReference.isDerived()) {
+                    nonContainmentReferences.add(setting);
                 }
             }
         }
-        return containmentReferences;
+        return nonContainmentReferences;
     }
 }

